sycl/source/device_selector.cpp

```diff
@@ -28,8 +29,51 @@ static bool isDeviceOfPreferredSyclBe(const device &Device) {
          backend::level0;
 }
 
-device device_selector::select_device() const {
-  vector_class<device> devices = device::get_devices();
+// return a device with the requested deviceType (and requested backend)
+// if no such device is found, heuristic is used to select a device.
+// 'deviceType' is the desired device type
+//   info::device_type::all means it relies on the heuristic to select a device
+// 'be' is a specific desired GPU backend choice when multiple backends are
+// found.
+device device_selector::select_device(info::device_type deviceType,
+                                      backend be) const {
+  // return if a requested deviceType is found
+  if (deviceType != info::device_type::all) {
+    if (deviceType == info::device_type::host) {
+      vector_class<device> devices;
+      devices.resize(1);
+      return devices[0];
+    }
+
+    const vector_class<detail::plugin> &plugins = RT::initialize();
+    for (unsigned int i = 0; i < plugins.size(); i++) {
+      pi_uint32 numPlatforms = 0;
+      plugins[i].call<detail::PiApiKind::piPlatformsGet>(0, nullptr,
+                                                         &numPlatforms);
+      if (numPlatforms) {
+        vector_class<RT::PiPlatform> piPlatforms(numPlatforms);
+        plugins[i].call<detail::PiApiKind::piPlatformsGet>(
+            numPlatforms, piPlatforms.data(), nullptr);
+        for (const auto &piPlatform : piPlatforms) {
+          platform pltf = detail::createSyclObjFromImpl<platform>(
+              std::make_shared<detail::platform_impl>(piPlatform, plugins[i]));
+          if (!pltf.is_host()) {
```

Comment:
Why this check is needed? Please, add a comment.

Reply:
HOST devices are not served by plugins. So, usually when you look for a host device, instead of calling plugin_XXX::piDevicesGet(), you just create a 'sycl::device' object whose constructor will create a host device. I will add a comment about this.